contracts/bridges/polygon/PolygonBridgeReceiver.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.15;
+
+import "../vendor/fx-portal/contracts/FxChild.sol";
+import "../BaseBridgeReceiver.sol";
+
+contract PolygonBridgeReceiver is IFxMessageProcessor, BaseBridgeReceiver {
+    error InvalidChild();
+
+    event NewFxChild(address indexed oldFxChild, address indexed newFxChild);
+
+    address public fxChild;
+
+    constructor(address _fxChild, address _initializer) BaseBridgeReceiver(_initializer) {
```

Comment:
I guess the benefit of making `initialize` publicly callable is we can get rid of this `_initializer` constructor arg. Something to consider 🤔

Reply:
I think the publicly-callable `initialize` function was for when the `initialize` function was basically just calling `timelock.acceptAdmin()`

If we made this version publicly-callable, then I think we could get in a bad state:

- PolygonBridgeReceiver is deployed (with no `mainnetTimelock` or `l2Timelock` set)
- Timelock is deployed, with PolygonBridgeReceiver as admin

at this point, if `initialize` was publicly-callable, then someone could take over the PolygonBridgeReceiver by setting their own contract as `mainnetTimelock`

Reply:
If we want this function to be pubicly-callable/to get rid of `initializer`, then I think the flow would have to be something like this:

- Timelock is deployed (with an EOA as admin)
- PolygonBridgeReceiver is deployed, setting both `mainnetTimelock` and `l2Timelock` in the constructor
- Timelock sets PolygonBridgeReceiver as pending admin
- PolygonBridgeReceiver.initialize function calls `timelock.acceptAdmin()`

Reply:
synced offline. going to remove the initializer (sounds like we don't mind that it's theoretically griefable)

- [ ] remove initializer

Reply:
I clarified this with Scott offline. Basically, the flow for setting up these contracts would not need to differ if we made `initialize` public. It would still be:

1. Deploy the BridgeReceiver
2. Deploy the Timelock with BridgeReceiver address as admin
3. Call BridgerReceiver.initialize with the mainnetTimelock and l2Timelock addresses

The difference is that `initialize` is public and we don't have to pass in an `initializer` address in the constructor (which I think is a lot cleaner, since `BaseBridgeReceiver` can have no constructor args). We will have an `initialized` boolean that flips to true after `initialize()` is called so it can only ever be called once.

In terms of griefing, the risk is minimal. Since all of these steps happen within seconds/minutes of each other in the deploy script, a griefer would have to call `initialize` in the moments between the `BridgeReceiver` being deployed and `initialize()` being called. Even if that happens, the deploy script would just fail when attempting to call `initialize()` (because it has already been initialized by the griefer) and we can just run the deploy script again.

Reply:
removed initializer